<a href="https://colab.research.google.com/github/MaureenGatu/Data-Analysis/blob/main/Global_data_on_confirmed_COVID_19_deaths.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# ***COVID-19 DEATHS DATA EXPLORATION***

## **Overview**

### Analysis Design



To get a proper picture and a good understanding of the data used for this analysis, the analysis was subdivided into the following sections.
1. ***Covid-19 Related deaths***

    a. Global analysis

    b. Anlysis by Continent

    c. Per country analysis
2. ***Covid-19 Vaccination***

    a. Global analysis

    b. Anlysis by Continent

    c. Per country analysis
3. ***Covid-19 Deaths vs. Vaccination Analysis***

    a. Global analysis

    b. Anlysis by Continent

    c. Per country analysis
4. ***Conclusion***

## Connecting to our Database

In [1]:
# We will first load the sql extension into our environment
%load_ext sql

# Then connect to our in memory sqlite database
# NB: This database will cease to exist as soon as the database connection is closed. 
%sql sqlite://

'Connected: @None'

## Importing Data from CSV files

Datasets used in this notebook:


1.   **Covid 19 Deaths**
2.   **Covid 19 Vaccinations** 

> The dataset is from Our World in Data. ([Dataset Download](https://ourworldindata.org/covid-deaths))






In [2]:
# Importing the pandas library
# We will use a function read_csv from pandas to read our datasets as shown
#
import pandas as pd 

In [3]:
# Loading our table from the respective CSV files 
with open('/content/drive/MyDrive/Data Science/Data Analysis/SQL/DATA/COVID_DEATHS_DATA.csv','r') as f:
    covid_deaths = pd.read_csv(f, index_col=0, encoding='utf-8')
# saving the urban center CSV file into a database
%sql DROP TABLE if EXISTS covid_deaths;
%sql PERSIST covid_deaths;

# Displaying the first  5 entries 
%sql SELECT * FROM covid_deaths LIMIT 5;

 * sqlite://
Done.
 * sqlite://
 * sqlite://
Done.


[('AFG', 'Asia', 'Afghanistan', '24/02/2020', 39835428.0, 5.0, 5.0, None, None, None, None, 0.126, 0.126, None, None, None, None, None, None, None, None, None, None, None, None, None),
 ('AFG', 'Asia', 'Afghanistan', '25/02/2020', 39835428.0, 5.0, 0.0, None, None, None, None, 0.126, 0.0, None, None, None, None, None, None, None, None, None, None, None, None, None),
 ('AFG', 'Asia', 'Afghanistan', '26/02/2020', 39835428.0, 5.0, 0.0, None, None, None, None, 0.126, 0.0, None, None, None, None, None, None, None, None, None, None, None, None, None),
 ('AFG', 'Asia', 'Afghanistan', '27/02/2020', 39835428.0, 5.0, 0.0, None, None, None, None, 0.126, 0.0, None, None, None, None, None, None, None, None, None, None, None, None, None),
 ('AFG', 'Asia', 'Afghanistan', '28/02/2020', 39835428.0, 5.0, 0.0, None, None, None, None, 0.126, 0.0, None, None, None, None, None, None, None, None, None, None, None, None, None)]

In [4]:
# Let's load our covid_vaccine dataset below
with open('/content/drive/MyDrive/Data Science/Data Analysis/SQL/DATA/COVID_VACCINATION_DATA.csv','r') as f:
    covid_vaccine = pd.read_csv(f, index_col=0, encoding='utf-8') 

# Save the houshold dataset into a database
%sql DROP TABLE if EXISTS covid_vaccine;
%sql PERSIST covid_vaccine;

# Display the first 5 entries from the database
%sql SELECT * FROM covid_vaccine LIMIT 5;

 * sqlite://
Done.
 * sqlite://
 * sqlite://
Done.


[('AFG', 'Asia', 'Afghanistan', '24/02/2020', None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, 8.33, 54.422, 18.6, 2.5810000000000004, 1.337, 1803.987, None, 597.029, 9.59, None, None, 37.746, 0.5, 64.83, 0.511, None, None, None, None),
 ('AFG', 'Asia', 'Afghanistan', '25/02/2020', None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, 8.33, 54.422, 18.6, 2.5810000000000004, 1.337, 1803.987, None, 597.029, 9.59, None, None, 37.746, 0.5, 64.83, 0.511, None, None, None, None),
 ('AFG', 'Asia', 'Afghanistan', '26/02/2020', None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, 8.33, 54.422, 18.6, 2.5810000000000004, 1.337, 1803.987, None, 597.029, 9.59, None, None, 37.746, 0.5, 64.83, 0.511, None, None, None, None),
 ('AFG', 'Asia', 'Afghanistan', '27/02/2020', None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, 8.33, 54.422, 18.6, 2.5810000000000004, 1.337, 1803.987, None, 597.029, 9.59, None, None, 37.746, 0.5, 64.83, 0.511, None, None, None, None),
 ('AFG', 'Asia', 'Afghanistan', '28/02/2020', None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, 8.33, 54.422, 18.6, 2.5810000000000004, 1.337, 1803.987, None, 597.029, 9.59, None, None, 37.746, 0.5, 64.83, 0.511, None, None, None, None)]

In [5]:
%%sql
--Lenght of covid_vaccine dataset
SELECT COUNT(*)
FROM covid_vaccine;

 * sqlite://
Done.


[(122175,)]

In [6]:
%%sql
--Lenght of covid_deaths dataset
SELECT COUNT(*)
FROM covid_deaths;

 * sqlite://
Done.


[(122175,)]

## **Data Exploration**

### ***Covid 19 Deaths***



#### **GLOBAL ANALYSIS**

In [7]:
%%sql
--Total confirmed  covid-19 related deaths worldwide
SELECT location,date,MAX(total_deaths) as total_deaths FROM covid_deaths;

 * sqlite://
Done.


[('World', '08/10/2021', 4842788.0)]

In [8]:
%%sql
--Highest number of reported covid-19 related deaths per million worldwide
SELECT location,date,MAX(total_deaths_per_million) as total_deaths_per_million FROM covid_deaths;

 * sqlite://
Done.


[('Peru', '08/10/2021', 5984.278)]

In [9]:
%%sql
-- Highest number of reported new cases of covid worldwide
SELECT location,date, MAX(new_cases) as Highest_reported_newcases FROM covid_deaths;

 * sqlite://
Done.


[('World', '28/04/2021', 905948.0)]

In [10]:
%%sql
-- Highest number of reported new deaths worldwide
SELECT location,date, MAX(new_deaths) as Highest_reported_newdeaths FROM covid_deaths;

 * sqlite://
Done.


[('World', '20/01/2021', 17977.0)]

In [11]:
%%sql
-- Highest number of reported new deaths worldwide
SELECT location,date, MAX(new_cases_per_million) as Highest_reported_newcases_permillion FROM covid_deaths;


 * sqlite://
Done.


[('Vatican', '12/10/2020', 8620.69)]

In [15]:
%%sql
-- Highest weekly ICU admissions globally
SELECT location, date, MAX(weekly_icu_admissions) as global_weekly_icu_admissions
FROM covid_deaths



 * sqlite://
Done.


[('France', '05/04/2020', 4002.4559999999997)]

##### ***Summary***
* **Total confirmed** covid-19 related **deaths** were **4.84 milion** as of 8th October 2021.

* Highest number of reported covid-19 related **deaths per million worldwide** was recorded in **Peru on 8th October 2021 which was 5984.278** deaths per a million people.

* Highest number of reported **new cases** of covid worldwide was recorded on **April 28th  2021 with 905,948** new reported cases.

* **Highest** number of reported **new deaths worldwide** was recorded on **20th January 2021 with 17,977** new deaths being recorded.

* Highest number of reported **new cases per million worldwide** was reocrded in **Vatican** with **8,620 new cases/million on 12th October 2020**.

* **Highest weekly ICU admissions** globally were recorded in **France** with **4,002** admissions in the first week of **April 2020**

---



---




#### **ANALYSIS PER CONTINENT**


In [39]:
%%sql
-- First we'll look at the total popuation per continent.
SELECT location, population
FROM covid_deaths
-- To exclude data that was aggregated per continent
WHERE continent IS  NULL AND location NOT IN ('World','International')
--Group by location to get highest per country
GROUP BY location 
--Order in descending order of the highest death count per location 
ORDER BY 2 desc;

 * sqlite://
Done.


[('Asia', 4679660580.0),
 ('Africa', 1373486472.0),
 ('Europe', 747747396.0),
 ('North America', 596581283.0),
 ('European Union', 447189915.0),
 ('South America', 434260138.0),
 ('Oceania', 43219954.0)]

In [44]:
%%sql
-- Per continent death count.
SELECT location, MAX(total_deaths) as total_deaths 
FROM covid_deaths
-- To exclude data that was aggregated per continent
WHERE continent IS  NULL AND location NOT IN ('World','International')
--Group by location to get highest per country
GROUP BY location 
--Order in descending order of the highest death count per location 
ORDER BY 2 desc;

 * sqlite://
Done.


[('Europe', 1245000.0),
 ('South America', 1159845.0),
 ('Asia', 1140531.0),
 ('North America', 1081518.0),
 ('European Union', 776919.0),
 ('Africa', 213531.0),
 ('Oceania', 2348.0)]

In [40]:
%%sql
-- Continent wide death count per population
SELECT location, MAX((total_deaths/population))*100 as death_count_per_population
FROM covid_deaths
-- To select data that was aggregated per continent except the aggregated world data.
WHERE continent IS  NULL AND location NOT IN ('World','International')
--Group by location to get highest per country
GROUP BY location 
--Order in descending order of the highest death count per location 
ORDER BY 2 desc;


 * sqlite://
Done.


[('South America', 0.2670853017598405),
 ('North America', 0.18128594222088593),
 ('European Union', 0.17373356910340879),
 ('Europe', 0.16650007832324168),
 ('Asia', 0.024372088114134122),
 ('Africa', 0.015546640200181016),
 ('Oceania', 0.005432675842274149)]

In [28]:
%%sql
-- Continent wide death count per population
SELECT location, MAX(total_cases) as total_cases
FROM covid_deaths
-- To exclude data that was aggregated per continent
WHERE continent IS  NULL AND location NOT IN ('World','International')
--Group by location to get highest per country
GROUP BY location 
--Order in descending order of the highest death count per location 
ORDER BY 2 desc;

 * sqlite://
Done.


[('Asia', 76889805.0),
 ('Europe', 60551949.0),
 ('North America', 53271010.0),
 ('European Union', 38363236.0),
 ('South America', 37951049.0),
 ('Africa', 8380407.0),
 ('Oceania', 203523.0)]

In [45]:
%%sql
-- Continent percentage_deaths_of_total_cases
SELECT location, MAX((total_deaths/total_cases))*100 as percentage_deaths_of_total_cases
FROM covid_deaths
-- To exclude data that was aggregated per continent
WHERE continent IS  NULL AND location NOT IN ('World','International')
--Group by location to get highest per country
GROUP BY location 
--Order in descending order of the highest death count per location 
ORDER BY 2 desc;

 * sqlite://
Done.


[('European Union', 11.122714802366561),
 ('Europe', 9.953974434570346),
 ('South America', 7.739426320555919),
 ('North America', 7.096774193548387),
 ('Africa', 5.460146827477714),
 ('Asia', 4.3056537645648225),
 ('Oceania', 3.7037037037037033)]

In [27]:
%%sql
-- Continent wide death count per population
SELECT location, MAX((total_cases/population))*100 as infected_pop_percentage
FROM covid_deaths
-- To exclude data that was aggregated per continent
WHERE continent IS  NULL AND location NOT IN ('World','International')
--Group by location to get highest per country
GROUP BY location 
--Order in descending order of the highest death count per location 
ORDER BY 2 desc;

 * sqlite://
Done.


[('North America', 8.9293800388974),
 ('South America', 8.73924306632998),
 ('European Union', 8.578734607644272),
 ('Europe', 8.097915061144526),
 ('Asia', 1.643063715531266),
 ('Africa', 0.6101557729794662),
 ('Oceania', 0.4709005474647196)]

In [29]:
%%sql
-- Highest_reported_new_deaths per continent
SELECT location,date, MAX(new_deaths) as Highest_reported_new_deaths
FROM covid_deaths
-- To exclude data that was aggregated per continent
WHERE continent IS  NULL AND location NOT IN ('World','International')
--Group by location to get highest per country
GROUP BY location 
--Order in descending order of the highest death count per location 
ORDER BY 3 desc;

 * sqlite://
Done.


[('Asia', '10/06/2021', 8524.0),
 ('Europe', '19/01/2021', 7557.0),
 ('North America', '05/02/2021', 6980.0),
 ('South America', '08/04/2021', 5919.0),
 ('European Union', '19/01/2021', 5020.0),
 ('Africa', '05/07/2021', 1513.0),
 ('Oceania', '03/09/2020', 59.0)]

In [26]:
%%sql
-- Highest_reported_newcases per continent
SELECT location,date, MAX(new_cases) as Highest_reported_newcases
FROM covid_deaths
-- To exclude data that was aggregated per continent
WHERE continent IS  NULL AND location NOT IN ('World','International')
--Group by location to get highest per country
GROUP BY location 
--Order in descending order of the highest death count per location 
ORDER BY 3 desc;

 * sqlite://
Done.


[('Asia', '06/05/2021', 525129.0),
 ('Europe', '02/11/2020', 346447.0),
 ('North America', '08/01/2021', 335286.0),
 ('European Union', '02/11/2020', 274748.0),
 ('South America', '23/06/2021', 186399.0),
 ('Africa', '23/08/2021', 49021.0),
 ('Oceania', '08/10/2021', 3108.0)]

In [35]:
%%sql
-- Highest_reported_newcases per continent
SELECT location,date,MAX(new_cases_per_million) as new_cases_per_million
FROM covid_deaths
-- To exclude data that was aggregated per continent
WHERE continent IS  NULL AND location NOT IN ('World','International')
--Group by location to get highest per country
GROUP BY location 
--Order in descending order of the highest death count per location 
ORDER BY 3 desc;

 * sqlite://
Done.


[('European Union', '02/11/2020', 614.388),
 ('North America', '08/01/2021', 562.012),
 ('Europe', '02/11/2020', 463.32099999999997),
 ('South America', '23/06/2021', 429.23400000000004),
 ('Asia', '06/05/2021', 112.215),
 ('Oceania', '08/10/2021', 71.911),
 ('Africa', '23/08/2021', 35.691)]

In [38]:
%%sql
-- Highest_new_cases_per_million
SELECT location, MAX((new_cases/population))*100 as infected_pop_percentage
FROM covid_deaths
-- To exclude data that was aggregated per continent
WHERE continent IS  NULL AND location NOT IN ('World','International')
--Group by location to get highest per country
GROUP BY location 
--Order in descending order of the highest death count per location 
ORDER BY 2 desc;

 * sqlite://
Done.


[('European Union', 0.06143877372547634),
 ('North America', 0.05620122681589393),
 ('Europe', 0.04633209046976073),
 ('South America', 0.04292335024312086),
 ('Asia', 0.01122151897606215),
 ('Oceania', 0.007191122878103942),
 ('Africa', 0.003569092306283742)]

##### ***Summary***

* **Asia** has the highest recorded **population size** with over **4.6 billion** people followed by **Africa** with over **1.3 billion** people.

* **Highest** recorded **total cases** were in **Asia** with over **76.89 million** cases reported followed by Europe with 66.55 million reported cases.

* The **most deaths** were reported in **Europe** with **1.245 million** recorded covid-19 related cases, followed by **South America** with over **1.59million** reported deaths. Asia reported aa total of 1.14 million  covid-19 related deaths.

* **South America** had **lost** over **0.26% of its population** to covid, **North America** had **lost 0.18%**,European Union 0.17% Europe 0.16% while Africa and Ocean reportes less than 0.1% deaths.

* **North and South America** had the highest **percentage of thier population infected** with covid-19 which were **8.9%** and **8.74%** of the total population respectively.

* Highest recoerded new cases and new deaths were all in the year 2021 except in oceania.

* **European Union** and **North America** had the highest number of **new cases per millio**n with **614**, and **562** cases per million respectively.

#### **ANALYSIS PER COUNTRY**

In [49]:
%%sql
--Selecting the data that we are going to be using ordered by date and location
SELECT location, date, total_cases,new_cases, total_deaths, population FROM covid_deaths ORDER BY 1,2 LIMIT 10;

 * sqlite://
Done.


[('Afghanistan', '01/01/2021', 52513.0, 183.0, 2201.0, 39835428.0),
 ('Afghanistan', '01/02/2021', 55059.0, 36.0, 2404.0, 39835428.0),
 ('Afghanistan', '01/03/2020', 5.0, 0.0, None, 39835428.0),
 ('Afghanistan', '01/03/2021', 55733.0, 19.0, 2444.0, 39835428.0),
 ('Afghanistan', '01/04/2020', 192.0, 26.0, 4.0, 39835428.0),
 ('Afghanistan', '01/04/2021', 56517.0, 63.0, 2489.0, 39835428.0),
 ('Afghanistan', '01/05/2020', 2171.0, 344.0, 64.0, 39835428.0),
 ('Afghanistan', '01/05/2021', 59939.0, 194.0, 2631.0, 39835428.0),
 ('Afghanistan', '01/06/2020', 15836.0, 656.0, 269.0, 39835428.0),
 ('Afghanistan', '01/06/2021', 72977.0, 1139.0, 2973.0, 39835428.0)]

In [50]:
%%sql
--Looking at total deaths vs total cases
SELECT location, date, total_cases, total_deaths, (total_deaths/total_cases)*100 as death_percentage
FROM covid_deaths WHERE total_cases AND total_deaths IS NOT NULL
ORDER BY 5 desc LIMIT 10;

 * sqlite://
Done.


[('Guyana', '12/03/2020', 1.0, 1.0, 100.0),
 ('Guyana', '13/03/2020', 1.0, 1.0, 100.0),
 ('Guyana', '14/03/2020', 1.0, 1.0, 100.0),
 ('Iran', '19/02/2020', 2.0, 2.0, 100.0),
 ('Peru', '06/03/2020', 1.0, 1.0, 100.0),
 ('Peru', '07/03/2020', 1.0, 1.0, 100.0),
 ('San Marino', '29/02/2020', 1.0, 1.0, 100.0),
 ('San Marino', '01/03/2020', 1.0, 1.0, 100.0),
 ('San Marino', '02/03/2020', 1.0, 1.0, 100.0),
 ('Sudan', '14/03/2020', 1.0, 1.0, 100.0)]

In [51]:
%%sql
--Looking at total deaths vs total cases
SELECT location, date, total_cases, total_deaths, (total_deaths/total_cases)*100 as death_percentage
FROM covid_deaths 
WHERE location like '%Kenya%' AND  total_cases AND total_deaths IS NOT NULL 
ORDER BY 5 desc LIMIT 5;

 * sqlite://
Done.


[('Kenya', '16/05/2020', 830.0, 50.0, 6.024096385542169),
 ('Kenya', '15/05/2020', 781.0, 45.0, 5.761843790012804),
 ('Kenya', '17/05/2020', 887.0, 50.0, 5.636978579481398),
 ('Kenya', '14/05/2020', 758.0, 42.0, 5.540897097625329),
 ('Kenya', '18/05/2020', 912.0, 50.0, 5.482456140350877)]

In [52]:
%%sql
--Looking at percentage of infection per country 
  --Ordered by highest infected population percentage
SELECT location, date, total_cases, total_deaths, (total_cases/population)*100 as infected_pop_percentage
FROM covid_deaths 
GROUP BY location
ORDER BY 5 desc LIMIT 5;

 * sqlite://
Done.


[('Seychelles', '08/10/2021', 21726.0, 118.0, 21.965423111919925),
 ('Montenegro', '08/10/2021', 134602.0, 1980.0, 21.43169901807337),
 ('Andorra', '08/10/2021', 15291.0, 130.0, 19.76756211702045),
 ('San Marino', '08/10/2021', 5466.0, 91.0, 16.07174360482211),
 ('Czechia', '08/10/2021', 1698061.0, 30496.0, 15.833396506129441)]

Seychelles, Montenegro and Andorra had the highest percentage of infeccted population which were 21.97%, 21.43% and 19.77% respectively as of 08/10/2021.

In [53]:
%%sql
--Looking at top percentage of infected population in Kenya.
SELECT location, total_cases, MAX(total_deaths) as highest_total_deaths, MAX((total_cases/population))*100 as infected_pop_percentage
FROM covid_deaths
WHERE continent IS NOT NULL
GROUP BY location  
ORDER BY 4 desc LIMIT 15;

 * sqlite://
Done.


[('Seychelles', 21726.0, 118.0, 21.965423111919925),
 ('Montenegro', 134602.0, 1980.0, 21.43169901807337),
 ('Andorra', 15291.0, 130.0, 19.76756211702045),
 ('San Marino', 5466.0, 91.0, 16.07174360482211),
 ('Czechia', 1698061.0, 30496.0, 15.833396506129441),
 ('Georgia', 628719.0, 9192.0, 15.797860832766089),
 ('Bahrain', 275538.0, 1390.0, 15.760383688107558),
 ('Maldives', 85394.0, 233.0, 15.708399249475738),
 ('Israel', 1302777.0, 7885.0, 14.821503983719268),
 ('Slovenia', 300084.0, 4899.0, 14.435978242411327),
 ('Serbia', 994708.0, 8631.0, 14.398896156233498),
 ('Cyprus', 120992.0, 558.0, 13.503496074240656),
 ('United States', 44290052.0, 712693.0, 13.303708801122054),
 ('Lithuania', 347548.0, 5179.0, 12.920662844413581),
 ('Luxembourg', 79018.0, 836.0, 12.447425545120302)]

Percentage of Kenyan population infected with covid 19 as at 08/10/2021 was 0.456%

In [54]:
%%sql
--Looking at coutries with the highest total death count.
SELECT location, total_cases, MAX(total_deaths) as maximum_deaths 
FROM covid_deaths 
WHERE continent IS NOT NULL
group by location
ORDER BY 3 desc LIMIT 5;

 * sqlite://
Done.


[('United States', 44290052.0, 712693.0),
 ('Brazil', 21550730.0, 600425.0),
 ('India', 33935309.0, 450375.0),
 ('Mexico', 3714392.0, 281610.0),
 ('Russia', 7602386.0, 210673.0)]

In [55]:
%%sql
-- Top 5 countries with highest death count per population
SELECT location, MAX((total_deaths/population))*100 as death_count_per_population
FROM covid_deaths
-- To exclude data that was aggregated per continent
WHERE continent IS NOT NULL
--Group by location to get highest per country
GROUP BY location 
--Order in descending order of the highest death count per location 
ORDER BY 2 desc LIMIT 5;


 * sqlite://
Done.


[('Peru', 0.5984277601990323),
 ('Bosnia and Herzegovina', 0.332867672000782),
 ('North Macedonia', 0.3264573543173853),
 ('Montenegro', 0.31526102179600063),
 ('Hungary', 0.3142463246933153)]

Peru had the highest death to population ratio at 0.60% percent followed by Bosmia and Herzegovina at 0.33%.

### ***Vaccination analysis***

#### Global Numbers

In [ ]:
%%sql
SELECT date, SUM(new_deaths	) as global_new_death_count
FROM covid_deaths
--
GROUP BY date 
--Order in descending order of the highest death count per location 
ORDER BY 2 desc LIMIT 5;

UsageError: Cell magic `%%sql` not found.


In [ ]:
%%sql
SELECT date, SUM(new_cases) as global_new_cases
FROM covid_deaths
--
GROUP BY date 
--Order in descending order of the highest death count per location 
ORDER BY 2 desc LIMIT 5;

UsageError: Cell magic `%%sql` not found.


In [ ]:
%%sql
-- Highest weekly ICU admissions globally
SELECT date, SUM(weekly_icu_admissions) as global_weekly_icu_admissions
FROM covid_deaths

GROUP BY date 
-- View top 5 ICU admissions
ORDER BY 2 desc LIMIT 5;

#### Analysis by continent

In [ ]:
%%sql
-- Continent wide death count per population
SELECT location, MAX((total_cases/population))*100 as infected_pop_percentage
FROM covid_deaths
-- To exclude data that was aggregated per continent
WHERE continent IS  NULL 
--Group by location to get highest per country
GROUP BY location 
--Order in descending order of the highest death count per location 
ORDER BY 2 desc;

 * sqlite://
Done.


[('North America', 8.9293800388974),
 ('South America', 8.73924306632998),
 ('European Union', 8.578734607644272),
 ('Europe', 8.097915061144526),
 ('World', 3.0127030018166945),
 ('Asia', 1.643063715531266),
 ('Africa', 0.6101557729794662),
 ('Oceania', 0.4709005474647196),
 ('International', None)]

In [ ]:
%%sql
-- Continent wide death count per population
SELECT location, MAX((total_deaths/population))*100 as death_count_per_population
FROM covid_deaths
-- To exclude data that was aggregated per continent
WHERE continent IS  NULL 
--Group by location to get highest per country
GROUP BY location 
--Order in descending order of the highest death count per location 
ORDER BY 2 desc;


 * sqlite://
Done.


[('South America', 0.2670853017598405),
 ('North America', 0.18128594222088593),
 ('European Union', 0.17373356910340879),
 ('Europe', 0.16650007832324168),
 ('World', 0.06149598823422537),
 ('Asia', 0.024372088114134122),
 ('Africa', 0.015546640200181016),
 ('Oceania', 0.005432675842274149),
 ('International', None)]

#### Analysis by country

In [ ]:
%%sql
--Selecting the data that we are going to be using ordered by date and location
SELECT location, date, total_cases,new_cases, total_deaths, population FROM covid_deaths ORDER BY 1,2 LIMIT 10;

UsageError: Cell magic `%%sql` not found.


In [ ]:
%%sql
--Looking at total deaths vs total cases
SELECT location, date, total_cases, total_deaths, (total_deaths/total_cases)*100 as death_percentage
FROM covid_deaths WHERE total_cases AND total_deaths IS NOT NULL
ORDER BY 5 desc LIMIT 10;

UsageError: Cell magic `%%sql` not found.


In [ ]:
%%sql
--Looking at total deaths vs total cases
SELECT location, date, total_cases, total_deaths, (total_deaths/total_cases)*100 as death_percentage
FROM covid_deaths 
WHERE location like '%Kenya%' AND  total_cases AND total_deaths IS NOT NULL 
ORDER BY 5 desc LIMIT 5;

In [ ]:
%%sql
--Looking at percentage of infection per country 
  --Ordered by highest infected population percentage
SELECT location, date, total_cases, total_deaths, (total_cases/population)*100 as infected_pop_percentage
FROM covid_deaths 
GROUP BY location
ORDER BY 5 desc LIMIT 5;

Seychelles, Montenegro and Andorra had the highest percentage of infeccted population which were 21.97%, 21.43% and 19.77% respectively as of 08/10/2021.

In [ ]:
%%sql
--Looking at top percentage of infected population in Kenya.
SELECT location, total_cases, MAX(total_deaths) as highest_total_deaths, MAX((total_cases/population))*100 as infected_pop_percentage
FROM covid_deaths
WHERE continent IS NOT NULL
GROUP BY location  
ORDER BY 4 desc LIMIT 15;

Percentage of Kenyan population infected with covid 19 as at 08/10/2021 was 0.456%

In [ ]:
%%sql
--Looking at coutries with the highest total death count.
SELECT location, total_cases, MAX(total_deaths) as maximum_deaths 
FROM covid_deaths 
WHERE continent IS NOT NULL
group by location
ORDER BY 3 desc LIMIT 5;

In [ ]:
%%sql
-- Top 5 countries with highest death count per population
SELECT location, MAX((total_deaths/population))*100 as death_count_per_population
FROM covid_deaths
-- To exclude data that was aggregated per continent
WHERE continent IS NOT NULL
--Group by location to get highest per country
GROUP BY location 
--Order in descending order of the highest death count per location 
ORDER BY 2 desc LIMIT 5;


Peru had the highest death to population ratio at 0.60% percent followed by Bosmia and Herzegovina at 0.33%.

In [ ]:
%%sql


### ***Covid Deaths vs Vaccinations***

#### Global Numbers

In [ ]:
%%sql
SELECT date, SUM(new_deaths	) as global_new_death_count
FROM covid_deaths
--
GROUP BY date 
--Order in descending order of the highest death count per location 
ORDER BY 2 desc LIMIT 5;

UsageError: Cell magic `%%sql` not found.


In [ ]:
%%sql
SELECT date, SUM(new_cases) as global_new_cases
FROM covid_deaths
--
GROUP BY date 
--Order in descending order of the highest death count per location 
ORDER BY 2 desc LIMIT 5;

UsageError: Cell magic `%%sql` not found.


In [ ]:
%%sql
-- Highest weekly ICU admissions globally
SELECT date, SUM(weekly_icu_admissions) as global_weekly_icu_admissions
FROM covid_deaths

GROUP BY date 
-- View top 5 ICU admissions
ORDER BY 2 desc LIMIT 5;

In [27]:
%%sql
-- Joining the tableson location and date
SELECT * FROM covid_deaths JOIN  covid_vaccine 
on covid_deaths.location = covid_vaccine.location 
and covid_deaths.date = covid_vaccine.date
LIMIT 10; 

 * sqlite://
Done.


[('AFG', 'Asia', 'Afghanistan', '24/02/2020', 39835428.0, 5.0, 5.0, None, None, None, None, 0.126, 0.126, None, None, None, None, None, None, None, None, None, None, None, None, None, 'AFG', 'Asia', 'Afghanistan', '24/02/2020', None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, 8.33, 54.422, 18.6, 2.5810000000000004, 1.337, 1803.987, None, 597.029, 9.59, None, None, 37.746, 0.5, 64.83, 0.511, None, None, None, None),
 ('AFG', 'Asia', 'Afghanistan', '25/02/2020', 39835428.0, 5.0, 0.0, None, None, None, None, 0.126, 0.0, None, None, None, None, None, None, None, None, None, None, None, None, None, 'AFG', 'Asia', 'Afghanistan', '25/02/2020', None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, 8.33, 54.422, 18.6, 2.5810000000000004, 1.337, 1803.987, None, 597.029, 9.59, None, None, 37.746, 0.5, 64.83, 0.511, None, None, None, None),
 ('AFG', 'Asia', 'Afghanistan', '26/02/2020', 39835428.0, 5.0, 0.0, None, None, None, None, 0.126, 0.0, None, None, None, None, None, None, None, None, None, None, None, None, None, 'AFG', 'Asia', 'Afghanistan', '26/02/2020', None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, 8.33, 54.422, 18.6, 2.5810000000000004, 1.337, 1803.987, None, 597.029, 9.59, None, None, 37.746, 0.5, 64.83, 0.511, None, None, None, None),
 ('AFG', 'Asia', 'Afghanistan', '27/02/2020', 39835428.0, 5.0, 0.0, None, None, None, None, 0.126, 0.0, None, None, None, None, None, None, None, None, None, None, None, None, None, 'AFG', 'Asia', 'Afghanistan', '27/02/2020', None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, 8.33, 54.422, 18.6, 2.5810000000000004, 1.337, 1803.987, None, 597.029, 9.59, None, None, 37.746, 0.5, 64.83, 0.511, None, None, None, None),
 ('AFG', 'Asia', 'Afghanistan', '28/02/2020', 39835428.0, 5.0, 0.0, None, None, None, None, 0.126, 0.0, None, None, None, None, None, None, None, None, None, None, None, None, None, 'AFG', 'Asia', 'Afghanistan', '28/02/2020', None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, 8.33, 54.422, 18.6, 2.5810000000000004, 1.337, 1803.987, None, 597.029, 9.59, None, None, 37.746, 0.5, 64.83, 0.511, None, None, None, None),
 ('AFG', 'Asia', 'Afghanistan', '29/02/2020', 39835428.0, 5.0, 0.0, 0.7140000000000001, None, None, 0.0, 0.126, 0.0, 0.018000000000000002, None, None, 0.0, None, None, None, None, None, None, None, None, None, 'AFG', 'Asia', 'Afghanistan', '29/02/2020', None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, 8.33, 54.422, 18.6, 2.5810000000000004, 1.337, 1803.987, None, 597.029, 9.59, None, None, 37.746, 0.5, 64.83, 0.511, None, None, None, None),
 ('AFG', 'Asia', 'Afghanistan', '01/03/2020', 39835428.0, 5.0, 0.0, 0.7140000000000001, None, None, 0.0, 0.126, 0.0, 0.018000000000000002, None, None, 0.0, None, None, None, None, None, None, None, None, None, 'AFG', 'Asia', 'Afghanistan', '01/03/2020', None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, 27.78, 54.422, 18.6, 2.5810000000000004, 1.337, 1803.987, None, 597.029, 9.59, None, None, 37.746, 0.5, 64.83, 0.511, None, None, None, None),
 ('AFG', 'Asia', 'Afghanistan', '02/03/2020', 39835428.0, 5.0, 0.0, 0.0, None, None, 0.0, 0.126, 0.0, 0.0, None, None, 0.0, None, None, None, None, None, None, None, None, None, 'AFG', 'Asia', 'Afghanistan', '02/03/2020', None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, 27.78, 54.422, 18.6, 2.5810000000000004, 1.337, 1803.987, None, 597.029, 9.59, None, None, 37.746, 0.5, 64.83, 0.511, None, None, None, None),
 ('AFG', 'Asia', 'Afghanistan', '03/03/2

In [37]:
%%sql
---Total vaccination vs population
SELECT covid_deaths.location, MAX(((covid_vaccine.people_vaccinated- covid_vaccine.people_fully_vaccinated)/covid_deaths.population))*100 as people_partially_vaccinated
FROM covid_deaths JOIN  covid_vaccine 
on covid_deaths.location = covid_vaccine.location 
and covid_deaths.date = covid_vaccine.date
WHERE covid_deaths.continent IS NULL
GROUP BY covid_deaths.location
ORDER BY 2 DESC
LIMIT 10;

 * sqlite://
Done.


[('South America', 24.634861834820306),
 ('Asia', 23.54395672858821),
 ('European Union', 20.135174336389046),
 ('World', 17.37371468729586),
 ('Oceania', 17.122102906449182),
 ('Europe', 15.206105512134743),
 ('North America', 11.909894598553807),
 ('Africa', 2.449838108052367),
 ('International', None)]

In [49]:
%%sql
---Total vaccination vs population
SELECT covid_deaths.location, MAX((covid_vaccine.people_fully_vaccinated/covid_deaths.population))*100 as people_fully_vaccinated
FROM covid_deaths JOIN  covid_vaccine 
on covid_deaths.location = covid_vaccine.location 
and covid_deaths.date = covid_vaccine.date
WHERE covid_deaths.continent IS NULL
GROUP BY covid_deaths.location
ORDER BY 2 DESC
LIMIT 10;

 * sqlite://
Done.


[('European Union', 63.513990694535224, 573299608.0),
 ('Europe', 53.2014887819148, 813382928.0),
 ('North America', 48.265529812137935, 635020739.0),
 ('South America', 43.78810403270309, 460836637.0),
 ('Asia', 38.19361377700602, 4354710132.0),
 ('Oceania', 36.73017328986514, 38276189.0),
 ('World', 34.829549871620955, 6463418984.0),
 ('Africa', 4.637739162239088, 159265365.0),
 ('International', None, None)]

In [93]:
%%sql
---Total vaccination vs population
SELECT covid_deaths.location,covid_deaths.date, SUM(covid_vaccine.new_vaccinations) OVER ( PARTITION BY covid_deaths.location order by covid_deaths.location, covid_deaths.date) as RollingPeopleVaccinated
FROM covid_deaths JOIN  covid_vaccine 
on covid_deaths.location = covid_vaccine.location 
and covid_deaths.date = covid_vaccine.date
WHERE covid_deaths.continent IS NOT  NULL
GROUP BY covid_deaths.location
--ORDER BY 2 DESC
LIMIT 10;

 * sqlite://
(sqlite3.OperationalError) near "(": syntax error
[SQL: ---Total vaccination vs population
SELECT covid_deaths.location,covid_deaths.date, SUM(covid_vaccine.new_vaccinations) OVER ( PARTITION BY covid_deaths.location order by covid_deaths.location, covid_deaths.date) as RollingPeopleVaccinated
FROM covid_deaths JOIN  covid_vaccine 
on covid_deaths.location = covid_vaccine.location 
and covid_deaths.date = covid_vaccine.date
WHERE covid_deaths.continent IS NOT  NULL
GROUP BY covid_deaths.location
--ORDER BY 2 DESC
LIMIT 10;]
(Background on this error at: https://sqlalche.me/e/14/e3q8)


In [78]:
%%sql
-- Rolling count
Select covid_deaths.continent, covid_deaths.location, covid_deaths.date, covid_deaths.population, covid_vaccine.new_vaccinations
SUM(covid_vaccine.new_vaccinations) OVER (Partition by covid_deaths.location 
Order by covid_deaths.location, covid_deaths.date) as RollingPeopleVaccinated
--,(RollingPeopleVaccinated/population)*100
From covid_deaths
Join covid_vaccine
	On covid_deaths.location = covid_vaccine.location
	and covid_deaths.date = covid_vaccine.date
where covid_deaths.continent is not null 
order by 2,3 LIMIT 10;

 * sqlite://
(sqlite3.OperationalError) near "(": syntax error
[SQL: -- Rolling count
Select covid_deaths.continent, covid_deaths.location, covid_deaths.date, covid_deaths.population, covid_vaccine.new_vaccinations
SUM(covid_vaccine.new_vaccinations) OVER (Partition by covid_deaths.location 
Order by covid_deaths.location, covid_deaths.date) as RollingPeopleVaccinated,(RollingPeopleVaccinated/population)*100
From covid_deaths
Join covid_vaccine
	On covid_deaths.location = covid_vaccine.location
	and covid_deaths.date = covid_vaccine.date
where covid_deaths.continent is not null 
order by 2,3 LIMIT 10;]
(Background on this error at: https://sqlalche.me/e/14/e3q8)


#### Analysis by continent

In [ ]:
%%sql
-- Continent wide death count per population
SELECT location, MAX((total_cases/population))*100 as infected_pop_percentage
FROM covid_deaths
-- To exclude data that was aggregated per continent
WHERE continent IS  NULL 
--Group by location to get highest per country
GROUP BY location 
--Order in descending order of the highest death count per location 
ORDER BY 2 desc;

 * sqlite://
Done.


[('North America', 8.9293800388974),
 ('South America', 8.73924306632998),
 ('European Union', 8.578734607644272),
 ('Europe', 8.097915061144526),
 ('World', 3.0127030018166945),
 ('Asia', 1.643063715531266),
 ('Africa', 0.6101557729794662),
 ('Oceania', 0.4709005474647196),
 ('International', None)]

In [ ]:
%%sql
-- Continent wide death count per population
SELECT location, MAX((total_deaths/population))*100 as death_count_per_population
FROM covid_deaths
-- To exclude data that was aggregated per continent
WHERE continent IS  NULL 
--Group by location to get highest per country
GROUP BY location 
--Order in descending order of the highest death count per location 
ORDER BY 2 desc;


 * sqlite://
Done.


[('South America', 0.2670853017598405),
 ('North America', 0.18128594222088593),
 ('European Union', 0.17373356910340879),
 ('Europe', 0.16650007832324168),
 ('World', 0.06149598823422537),
 ('Asia', 0.024372088114134122),
 ('Africa', 0.015546640200181016),
 ('Oceania', 0.005432675842274149),
 ('International', None)]

#### Analysis by country

In [ ]:
%%sql
--Selecting the data that we are going to be using ordered by date and location
SELECT location, date, total_cases,new_cases, total_deaths, population FROM covid_deaths ORDER BY 1,2 LIMIT 10;

UsageError: Cell magic `%%sql` not found.


In [ ]:
%%sql
--Looking at total deaths vs total cases
SELECT location, date, total_cases, total_deaths, (total_deaths/total_cases)*100 as death_percentage
FROM covid_deaths WHERE total_cases AND total_deaths IS NOT NULL
ORDER BY 5 desc LIMIT 10;

UsageError: Cell magic `%%sql` not found.


In [ ]:
%%sql
--Looking at total deaths vs total cases
SELECT location, date, total_cases, total_deaths, (total_deaths/total_cases)*100 as death_percentage
FROM covid_deaths 
WHERE location like '%Kenya%' AND  total_cases AND total_deaths IS NOT NULL 
ORDER BY 5 desc LIMIT 5;

In [ ]:
%%sql
--Looking at percentage of infection per country 
  --Ordered by highest infected population percentage
SELECT location, date, total_cases, total_deaths, (total_cases/population)*100 as infected_pop_percentage
FROM covid_deaths 
GROUP BY location
ORDER BY 5 desc LIMIT 5;

Seychelles, Montenegro and Andorra had the highest percentage of infeccted population which were 21.97%, 21.43% and 19.77% respectively as of 08/10/2021.

In [ ]:
%%sql
--Looking at top percentage of infected population in Kenya.
SELECT location, total_cases, MAX(total_deaths) as highest_total_deaths, MAX((total_cases/population))*100 as infected_pop_percentage
FROM covid_deaths
WHERE continent IS NOT NULL
GROUP BY location  
ORDER BY 4 desc LIMIT 15;

Percentage of Kenyan population infected with covid 19 as at 08/10/2021 was 0.456%

In [ ]:
%%sql
--Looking at coutries with the highest total death count.
SELECT location, total_cases, MAX(total_deaths) as maximum_deaths 
FROM covid_deaths 
WHERE continent IS NOT NULL
group by location
ORDER BY 3 desc LIMIT 5;

In [ ]:
%%sql
-- Top 5 countries with highest death count per population
SELECT location, MAX((total_deaths/population))*100 as death_count_per_population
FROM covid_deaths
-- To exclude data that was aggregated per continent
WHERE continent IS NOT NULL
--Group by location to get highest per country
GROUP BY location 
--Order in descending order of the highest death count per location 
ORDER BY 2 desc LIMIT 5;


Peru had the highest death to population ratio at 0.60% percent followed by Bosmia and Herzegovina at 0.33%.

In [ ]:
%%sql


In [ ]:
%%sql



In [ ]:
%%sql


In [ ]:
%%sql


# **Conclussion**